lib/ex_unit/lib/ex_unit/server.ex

```diff
@@ -51,34 +57,39 @@ defmodule ExUnit.Server do
     state = %{
       loaded: System.monotonic_time(),
       waiting: nil,
-      async_modules: :queue.new(),
+      async_partitions: %{},
+      async_partition_keys: :queue.new(),
```

Comment:
The idea with this split between the group keys queue (to be renamed from partition keys) and the mapping from a group key to the modules was to make it faster to access the modules by group key.

But after completing the functionality I'm not sure it's needed and we could just have a single field `async_modules` which would be a queue where each item is tuple `{group_key, modules}`.

To make it consistent, we can also store the `sync_modules` the same way; only the modules list will always be single element.

WDYT @josevalim

Reply:
I've tried to make it simpler in https://github.com/elixir-lang/elixir/pull/13897/commits/3b9d737354a2964bd3ac188857526d9603181146.